Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -98,20 +102,15 @@ final class ConfigGenerator: ConfigGenerating {
                                             fileElements: ProjectFileElements,
                                             pbxproj: PBXProj,
                                             configurationList: XCConfigurationList) throws {
-        let variant: BuildSettingsProvider.Variant = (buildConfiguration.variant == .debug) ? .debug : .release
-        let defaultConfigSettings = BuildSettingsProvider.projectDefault(variant: variant)
-        let defaultSettingsAll = BuildSettingsProvider.projectDefault(variant: .all)
-
-        var settings: [String: Any] = [:]
-        extend(buildSettings: &settings, with: defaultSettingsAll)
-        extend(buildSettings: &settings, with: project.settings.base)
-        extend(buildSettings: &settings, with: defaultConfigSettings)
+        var settings = defaultSettingsProvider.defaultProjectSettings(project: project,
```

Comment:
Nitpick: Including `default` in the name of the method is a bit redundant considering that the name of the instance is `defaultSettingsProvider`.

Reply:
The method is a part of the protocol, as the name of the variable is often out of the class author control, I usually try to be very explicit to reduce potential questions.

Eg.
```swift
provider.projectSettings(project: project) 
```

but here we're both authors and clients so making it less verbose makes sense to me :)